scripts/release/update-helm-charts.bash

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$DIR/../.."
+
+HELM_CHARTS_PATH="$(pwd)/../helm-charts"
+CHARTS_FILE="charts/telegraf-operator/Chart.yaml"
+
+main() {
+  local version="${1:?Version required}"
+  local branch="${2:-telegraf-operator-v$(echo "${version}" | sed 's#\.#-#g')}"
+
+  if [ ! -f "${HELM_CHARTS_PATH}/${CHARTS_FILE}" ] ; then
+    echo >&2 "Unable to find ${HELM_CHARTS_PATH}/${CHARTS_FILE}; exiting..."
+    exit 1
+  fi
+
+  cd "${HELM_CHARTS_PATH}"
+  if [ "$(git rev-parse --abbrev-ref HEAD)" != "master" ] ; then
+    echo "helm-charts repo at $(pwd) must be checked out as \"master\""
+    exit 1
+  fi
+
+  if [ ! -z "$(git status --porcelain)" ]; then
+    echo "helm-charts repo at $(pwd) has local changes"
+    exit 1
+  fi
+
+  git pull
+  git checkout -b "${branch}"
+
+  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
+    -i "${CHARTS_FILE}"
+
+  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
+    -i "${CHARTS_FILE}"
+
```

Comment:
so it works also on BSD sed (e.g. on macos)

```suggestion
  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
    -i '' "${CHARTS_FILE}"

  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
    -i '' "${CHARTS_FILE}"

```

Reply:
Just out of curiosity - why is the `''` argument needed on macos?

Reply:
Ah I see, `-i` is something else on macos than Linux.

It seems there's no (easy?) way to call sed with in-place replacement on macos and linux.

This is why we can't just have nice things - anything beyond basic functionality is just different between OSes.

Reply:
Ok, changed it so it is not using the in-place editing of files, but a temporary file.

I am not happy with that, but it should work on both OSes.

Reply:
the question is why is this optional on linux:

linux man page:

```
    -i[SUFFIX], --in-place[=SUFFIX]

              edit files in place (makes backup if SUFFIX supplied)
```

bsd man page:

```
     -i extension
             Edit files in-place, saving backups with the specified extension.  If a zero-length extension is given, no backup will be saved.  It is not recommended to give a zero-length extension
             when in-place editing files, as you risk corruption or partial content in situations where disk space is exhausted, etc.
```

Reply:
It is worse, on Linux it is optional, but if you want to provide value, it has to be non-empty and same argument.

So there is no way to invoke sed with in-place in a consistent. The only answer I found in https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux is to do `-i.bak` and then have a leftover `.bak` file or having to remove it.

Maybe that's better than my current approach?

Reply:
(I don't like keeping tmp .bkp files unless we also clean up with a trap)